tests/foreman/virtwho/cli/test_esx_sca.py

```diff
@@ -0,0 +1,543 @@
+"""Test class for Virtwho Configure CLI
+
+:Requirement: Virt-whoConfigurePlugin
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: Virt-whoConfigurePlugin
+
+:Team: Phoenix
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+import re
+
+import pytest
+import requests
+from fauxfactory import gen_string
+
+from robottelo.cli.user import User
+from robottelo.config import settings
+from robottelo.utils.virtwho import create_http_proxy
+from robottelo.utils.virtwho import deploy_configure_by_command
+from robottelo.utils.virtwho import deploy_configure_by_command_check
+from robottelo.utils.virtwho import deploy_configure_by_script
+from robottelo.utils.virtwho import ETC_VIRTWHO_CONFIG
+from robottelo.utils.virtwho import get_configure_command
+from robottelo.utils.virtwho import get_configure_file
+from robottelo.utils.virtwho import get_configure_option
+from robottelo.utils.virtwho import hypervisor_json_create
+from robottelo.utils.virtwho import virtwho_package_locked
+
+
+@pytest.fixture()
+def form_data(target_sat, module_sca_manifest_org):
+    form = {
+        'name': gen_string('alpha'),
+        'debug': 1,
+        'interval': '60',
+        'hypervisor-id': 'hostname',
+        'hypervisor-type': settings.virtwho.esx.hypervisor_type,
+        'hypervisor-server': settings.virtwho.esx.hypervisor_server,
+        'organization-id': module_sca_manifest_org.id,
+        'filtering-mode': 'none',
+        'satellite-url': target_sat.hostname,
+        'hypervisor-username': settings.virtwho.esx.hypervisor_username,
+        'hypervisor-password': settings.virtwho.esx.hypervisor_password,
+    }
+    return form
+
+
+@pytest.fixture()
+def virtwho_config(form_data, target_sat):
+    virtwho_config = target_sat.cli.VirtWhoConfig.create(form_data)['general-information']
+    yield virtwho_config
+    target_sat.cli.VirtWhoConfig.delete({'name': virtwho_config['name']})
+    assert not target_sat.cli.VirtWhoConfig.exists(search=('name', form_data['name']))
```

Comment:
We can also add this  teardown in other modules where virtwho_config is defined and remove it from every test's teardown, what do you think?

Reply:
> We can also add this teardown in other modules where virtwho_config is defined and remove it from every test's teardown, what do you think?

Yes, I really agree with you, and I also plan to do this tasks in the future in optimized the cases, also will  use fixture to parameterize some similar cases.

Reply:
Great, Thanks!